tests/test_deploy.py

```diff
@@ -17,101 +17,28 @@
 """Juju GUI deploy module tests."""
 
 import os
-import shutil
-import tempfile
 import unittest
 
 import mock
 
-import deploy
-from deploy import (
-    juju_deploy,
-    setup_repository,
-)
+from deploy import juju_deploy
 
 
 def contains(needle, haystack):
     """Does the sequence (haystack) contain the given subsequence (needle)?"""
-    for i in xrange(len(haystack)-len(needle)+1):
-        for j in xrange(len(needle)):
+    for i in range(len(haystack)-len(needle)+1):
+        for j in range(len(needle)):
             if haystack[i+j] != needle[j]:
```

Comment:
For two reasons, not particularly strong:
1) xrange is no longer present in Python3, so using range is better in the unlikely case we'll switch the charm to Python3;
2) for such small ranges, a generator is probably overkill anyway, and slightly slower.